### PR TITLE
Upgrade to 2.0

### PR DESCRIPTION
**Pull request by TheMilek**
Screenshot 2024-10-16 at 08 45 07

Screenshot 2024-10-11 at 14 20 45
Screenshot 2024-10-11 at 14 21 07

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -5,26 +5,27 @@
...
-        "symfony/dependency-injection": "^5.4.21 || ^6.4",
-        "symfony/form": "^5.4.21 || ^6.4",
-        "symfony/framework-bundle": "^5.4.21  || ^6.4",
-        "symfony/http-foundation": "^5.4.21  || ^6.4",
-        "symfony/http-kernel": "^5.4.21  || ^6.4",
-        "symfony/messenger": "^5.4.21 || ^6.4",
-        "symfony/options-resolver": "^5.4.21  || ^6.4",
-        "symfony/routing": "^5.4.21  || ^6.4"
+        "sylius/sylius": "2.0.x-dev",
+        "sylius/twig-hooks": "^0.4",
```
Comment (Wojdylak) [inline composer.json:14]: Is it necessary?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -204,11 +191,16 @@ jobs:
 
             -
                 name: Run Behat
-                run: vendor/bin/behat --colors --strict -vvv --no-interaction -f progress || vendor/bin/behat --colors --strict -vvv --no-interaction -f progress --rerun
+                run: |
+                    if [ "${{ matrix.wkhtmltopdf }}" != "false" ]; then
+                        vendor/bin/behat --colors --strict -vvv --no-interaction -f progress --tags="~@pdf_disabled" || vendor/bin/behat --colors --strict -vvv --no-interaction -f progress --rerun;
```
Comment (Wojdylak) [inline .github/workflows/build.yml:196]: How did it work before?

File: src/Resources/config/app/twig_hooks/admin/invoice/show.yaml [hunk 3]
```diff
@@ -0,0 +1,198 @@
+sylius_twig_hooks:
+    hooks:
+        'sylius_admin.invoice.show.content':
+            page_body:
```
Comment (Wojdylak) [inline src/Resources/config/app/twig_hooks/admin/invoice/show.yaml:4]:
```suggestion
            sections:
```

File: src/templates/admin/invoice/show/content/page_body/buyer.html.twig [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{% set invoice = hookable_metadata.context.resource %}
+{% set billingData = invoice.billingData %}
```
Comment (Wojdylak) [inline src/templates/admin/invoice/show/content/page_body/buyer.html.twig:2]:
```suggestion
{% set billing_data = invoice.billingData %}
```

File: src/templates/admin/invoice/show/content/page_body/buyer/company.html.twig [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{% set billingData = hookable_metadata.context.resource.billingData %}
```
Comment (Wojdylak) [inline src/templates/admin/invoice/show/content/page_body/buyer/company.html.twig:1]:
```suggestion
{% set billing_data = hookable_metadata.context.resource.billingData %}
```
In all places.

File: src/Resources/config/app/twig_hooks/admin/invoice/show.yaml [hunk 6]
```diff
@@ -0,0 +1,198 @@
+sylius_twig_hooks:
+    hooks:
+        'sylius_admin.invoice.show.content':
```
Comment (GSadee) [inline src/Resources/config/app/twig_hooks/admin/invoice/show.yaml:3]:
```suggestion
        'sylius_invoicing.admin.show.content':
```
wdyt?

File: src/Resources/config/app/twig_hooks/admin/order/show.yaml [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+        'sylius_admin.order.show.content.sections.invoices.table.body.actions':
+            download:
+                template: '@SyliusInvoicingPlugin/admin/shared/action/download.html.twig'
+                enabled: '%sylius_invoicing.pdf_generator.enabled%'
+                priority: 100
+            resend:
+                template: '@SyliusInvoicingPlugin/admin/shared/action/resend.html.twig'
+                priority: 0
+
```
Comment (GSadee) [inline src/Resources/config/app/twig_hooks/admin/order/show.yaml:59]:
```suggestion
```

File: src/Resources/config/config.yml [hunk 8]
```diff
@@ -11,7 +14,7 @@ sylius_mailer:
     emails:
         invoice_generated:
             subject: sylius.emails.invoice_generated.subject
-            template: "@SyliusInvoicingPlugin/Invoice/Email/invoiceGenerated.html.twig"
+            template: "@SyliusInvoicingPlugin/admin/invoice/email/invoiceGenerated.html.twig"
```
Comment (GSadee) [inline src/Resources/config/config.yml:17]:
```suggestion
            template: "@SyliusInvoicingPlugin/admin/invoice/email/invoice_generated.html.twig"
```

File: src/templates/admin/invoice/show/content/page_body/seller.html.twig [hunk 9]
```diff
@@ -0,0 +1,21 @@
+{% set invoice = hookable_metadata.context.resource %}
+{% set shopBillingData = invoice.shopBillingData %}
```
Comment (GSadee) [inline src/templates/admin/invoice/show/content/page_body/seller.html.twig:2]: Use snake_case notation here and in other templates

File: .github/workflows/build.yml [hunk 10]
```diff
@@ -12,50 +12,38 @@ on:
...
-                        php: "8.2"
+                        php: "8.3"
                         symfony: "^6.4"
-                        sylius: "~1.12.0"
+                        sylius: "2.0.x-dev"
                         node: "20.x"
-                        mysql: "8.0"
+                        mysql: "8.4"
                         wkhtmltopdf: false
                         state_machine_adapter: "winzou_state_machine"
```
[the remark this answers is not in the document]
Comment (TheMilek) [reply .github/workflows/build.yml]: I had a problem with that earlier, somehow it works now, so yeah maybe i got some issue in other place 🤷🏽

File: .github/workflows/build.yml [hunk 11]
```diff
@@ -12,50 +12,38 @@ on:
 
 jobs:
     tests:
-        runs-on: ubuntu-latest
+        runs-on: ubuntu-22.04
```
[the remark this answers is not in the document]
Comment (TheMilek) [reply .github/workflows/build.yml]: there is an issue with wkhtmltopdf https://github.com/Sylius/InvoicingPlugin/actions/runs/11360282465/job/31597760279?pr=323

File: src/templates/admin/invoice/show/content/header/title_block/actions/back_to_index.html.twig [hunk 12]
```diff
@@ -0,0 +1,5 @@
+{% from '@SyliusAdmin/shared/helper/icon.html.twig' import icon %}
+
+<a {{ sylius_test_html_attribute('back') }}href="{{ path('sylius_invoicing_plugin_admin_invoice_index') }}" class="btn">
```
Comment (Wojdylak) [inline src/templates/admin/invoice/show/content/header/title_block/actions/back_to_index.html.twig:3]:
```suggestion
<a {{ sylius_test_html_attribute('back') }} href="{{ path('sylius_invoicing_plugin_admin_invoice_index') }}" class="btn">
```

File: src/templates/admin/invoice/show/content/sections/card/footer.html.twig [hunk 13]
```diff
@@ -0,0 +1,12 @@
+<div class="card-footer text-end">
+    <div class="d-inline-block" {{ sylius_test_html_attribute('invoice-is-paid') }}>
+        <strong>{{ 'sylius_invoicing_plugin.ui.payment.paid'|trans }}: </strong>
+        <span>
+            {% if hookable_metadata.context.resource.paymentState() is constant('Sylius\\InvoicingPlugin\\Entity\\InvoiceInterface::PAYMENT_STATE_COMPLETED') %}
```
Comment (Wojdylak) [inline src/templates/admin/invoice/show/content/sections/card/footer.html.twig:5]:
```suggestion
            {% if hookable_metadata.context.resource.paymentState is constant('Sylius\\InvoicingPlugin\\Entity\\InvoiceInterface::PAYMENT_STATE_COMPLETED') %}
```

**Review by Wojdylak — APPROVED** [review]

File: UPGRADE.md [hunk 14]
```diff
@@ -1,3 +1,10 @@
+### UPGRADE FROM 0.24.X TO 1.0.0
```
Comment (GSadee) [inline UPGRADE.md:1]: We will probably release support for Sylius 2.0 in version 2.0 of this plugin, and release earlier 1.0, but we could adjust it later 🖖🏻

File: UPGRADE.md [hunk 15]
```diff
@@ -1,3 +1,10 @@
...
+
+1. Support for Sylius 2.0 has been added, it is now the recommended Sylius version to use with InvoicingPlugin.
+2. Support for Sylius 1.13 and 1.12 has been dropped, upgrade your application to [Sylius 2.0](https://github.com/Sylius/Sylius/blob/2.0/UPGRADE-2.0.md).
+3. Support for PHP has been increased to 8.2
```
Comment (GSadee) [inline UPGRADE.md:5]:
```suggestion
3. The minimum supported PHP version has been changed to 8.2
```
🤔

File: src/Resources/config/config.yml [hunk 16]
```diff
@@ -1,3 +1,7 @@
+imports:
+    - { resource: "@SyliusInvoicingPlugin/Resources/config/app/winzou_state_machine.php" }
+    - { resource: 'app/twig_hooks/**/*.yaml' }
```
Comment (GSadee) [inline src/Resources/config/config.yml:3]: Make these imports consistent with each other

File: src/Resources/config/services/twig.xml [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+  ~
+  ~ For the full copyright and license information, please view the LICENSE
+  ~ file that was distributed with this source code.
+  -->
+
+<container xmlns="http://symfony.com/schema/dic/services" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+    <services>
+        <service
+                id="sylius_invoicing.twig.component.invoice.list"
+                class="Sylius\InvoicingPlugin\Twig\Component\Invoice\ListComponent"
```
Comment (GSadee) [inline src/Resources/config/services/twig.xml:16]: Wrong indentation

**Review by GSadee — APPROVED** [review]

**Pull request merged**